packages/wallet-sdk/src/errors/utils.ts

```diff
@@ -0,0 +1,150 @@
+ * no corresponding message.
+ */
+export function getMessageFromCode(
+  code: number | undefined,
+  fallbackMessage: string = FALLBACK_MESSAGE
+): string {
+  if (code && Number.isInteger(code)) {
+    const codeString = code.toString();
+
+    if (hasKey(errorValues, codeString)) {
+      return errorValues[codeString as ErrorValueKey].message;
+    }
+    if (isJsonRpcServerError(code)) {
+      return JSON_RPC_SERVER_ERROR_MESSAGE;
+    }
+  }
+  return fallbackMessage;
+}
+
+/**
+ * Returns whether the given code is valid.
+ * A code is only valid if it has a message.
+ */
+export function isValidCode(code: number): boolean {
+  if (!Number.isInteger(code)) {
+    return false;
+  }
+
+  const codeString = code.toString();
+  if (errorValues[codeString as ErrorValueKey]) {
+    return true;
+  }
+
+  if (isJsonRpcServerError(code)) {
+    return true;
+  }
+  return false;
+}
+
+/**
+ * Returns the error code from an error object.
+ */
+export function getErrorCode(error: unknown): number | undefined {
+  if (typeof error === 'number') {
+    return error;
+  } else if (isErrorWithCode(error)) {
+    return error.code ?? error.errorCode;
+  }
+
+  return undefined;
+}
+
+interface ErrorWithCode {
+  code?: number;
+  errorCode?: number;
+}
+
+function isErrorWithCode(error: unknown): error is ErrorWithCode {
+  return (
+    typeof error === 'object' &&
+    error !== null &&
+    (typeof (error as ErrorWithCode).code === 'number' ||
+      typeof (error as ErrorWithCode).errorCode === 'number')
+  );
+}
+
+/**
+ * Serializes the given error to an Ethereum JSON RPC-compatible error object.
+ * Merely copies the given error's values if it is already compatible.
+ * If the given error is not fully compatible, it will be preserved on the
+ * returned object's data.originalError property.
+ */
+
+export interface SerializedEthereumRpcError {
+  code: number; // must be an integer
+  message: string;
+  data?: unknown;
+  stack?: string;
+}
+
+export function serialize(
+  error: unknown,
+  { shouldIncludeStack = false } = {}
+): SerializedEthereumRpcError {
+  const serialized: Partial<SerializedEthereumRpcError> = {};
+
+  if (
+    error &&
+    typeof error === 'object' &&
+    !Array.isArray(error) &&
+    hasKey(error as Record<string, unknown>, 'code') &&
+    isValidCode((error as SerializedEthereumRpcError).code)
+  ) {
+    const _error = error as Partial<SerializedEthereumRpcError>;
+    serialized.code = _error.code;
+
+    if (_error.message && typeof _error.message === 'string') {
+      serialized.message = _error.message;
+
+      if (hasKey(_error, 'data')) {
+        serialized.data = _error.data;
+      }
+    } else {
+      serialized.message = getMessageFromCode((serialized as SerializedEthereumRpcError).code);
+
+      serialized.data = { originalError: assignOriginalError(error) };
+    }
+  } else {
+    serialized.code = standardErrorCodes.rpc.internal;
+
+    const message = (error as any)?.message;
```

Comment:
We should check the error using `instanceof` vs using any

```suggestion
let message = '';
if (error instanceof Error) {
 message =  error.message
}
```

Reply:
this is the code forked from eth-rpc-errors package.
btw, i don't think we should check whether it's Error instance. this code should be able to process any objects that has `message` prop even though they are not `Error`. 
is there better way to do that?

Reply:
Hmm.. we could maybe use an unknown vs any

```ts
message = (error as unknown as { message?: string })?.message
```

or something else could be

```ts
func hasMsgProperty(obj: unknown): obj is {message: string} {
  return typeof obj === 'object' && obj !== null && 'message' in obj;
}

const message = hasMsgProperty(error) ? error.message : FALLBACK_MESSAGE
```

Reply:
Option 2 is probably the best since it doesn't typecast

Reply:
thanks for the examples!